<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>R. Reucher</author>
 <class>MESSDeviceConfigurator</class>
 <widget class="QWidget" name="MESSDeviceConfigurator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>566</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MESS device configuration</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QSplitter" name="vSplitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QGroupBox" name="groupBoxActiveDeviceConfiguration">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Active device configuration</string>
      </property>
      <layout class="QGridLayout" name="gridLayoutActiveDeviceConfigs">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="frameConfiguration">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayoutFrameConfiguration" columnstretch="0,1,0,0,0,0">
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QToolButton" name="toolButtonConfiguration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Device configuration menu</string>
            </property>
            <property name="statusTip">
             <string>Device configuration menu</string>
            </property>
            <property name="icon">
             <iconset resource="qmc2.qrc">
              <normaloff>:/data/img/work.png</normaloff>:/data/img/work.png</iconset>
            </property>
            <property name="popupMode">
             <enum>QToolButton::MenuButtonPopup</enum>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonIconOnly</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEditConfigurationName">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Name of device configuration</string>
            </property>
            <property name="statusTip">
             <string>Name of device configuration</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="toolButtonNewConfiguration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Create a new device configuration</string>
            </property>
            <property name="statusTip">
             <string>Create a new device configuration</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="qmc2.qrc">
              <normaloff>:/data/img/filenew.png</normaloff>:/data/img/filenew.png</iconset>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QToolButton" name="toolButtonCloneConfiguration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Clone current device configuration</string>
            </property>
            <property name="statusTip">
             <string>Clone current device configuration</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="qmc2.qrc">
              <normaloff>:/data/img/editcopy.png</normaloff>:/data/img/editcopy.png</iconset>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QToolButton" name="toolButtonRemoveConfiguration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Remove current device configuration from list of available configurations</string>
            </property>
            <property name="statusTip">
             <string>Remove current device configuration from list of available configurations</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="qmc2.qrc">
              <normaloff>:/data/img/editdelete.png</normaloff>:/data/img/editdelete.png</iconset>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QToolButton" name="toolButtonSaveConfiguration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Save current device configuration to list of available configurations</string>
            </property>
            <property name="statusTip">
             <string>Save current device configuration to list of available configurations</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="qmc2.qrc">
              <normaloff>:/data/img/filesaveas.png</normaloff>:/data/img/filesaveas.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTabWidget" name="tabWidgetDeviceSetup">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tabDeviceMap">
          <attribute name="title">
           <string>Device mappings</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayoutDeviceMap">
           <property name="margin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTreeWidget" name="treeWidgetDeviceSetup">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="contextMenuPolicy">
              <enum>Qt::CustomContextMenu</enum>
             </property>
             <property name="toolTip">
              <string>Device setup of current configuration</string>
             </property>
             <property name="statusTip">
              <string>Device setup of current configuration</string>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="rootIsDecorated">
              <bool>false</bool>
             </property>
             <property name="uniformRowHeights">
              <bool>true</bool>
             </property>
             <column>
              <property name="text">
               <string>Device instance</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Brief name</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Type</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Tag</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Extensions</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>File</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabSlotOptions">
          <attribute name="title">
           <string>Slot options</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayoutSlotOptions">
           <property name="margin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTreeWidget" name="treeWidgetSlotOptions">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="contextMenuPolicy">
              <enum>Qt::CustomContextMenu</enum>
             </property>
             <property name="toolTip">
              <string>Available slot options</string>
             </property>
             <property name="statusTip">
              <string>Available slot options</string>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="rootIsDecorated">
              <bool>false</bool>
             </property>
             <property name="uniformRowHeights">
              <bool>true</bool>
             </property>
             <column>
              <property name="text">
               <string>Slot</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Option</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabFileChooser">
          <attribute name="title">
           <string>File chooser</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayoutFileChooser">
           <property name="margin">
            <number>0</number>
           </property>
           <property name="spacing">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QFrame" name="frameFileChooser">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayoutFrameFileChooser">
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="labelDeviceInstanceChooser">
                <property name="text">
                 <string>Device instance</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBoxDeviceInstanceChooser">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Select the device instance the file is mapped to</string>
                </property>
                <property name="statusTip">
                 <string>Select the device instance the file is mapped to</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBoxChooserAutoSelect">
                <property name="toolTip">
                 <string>Automatically select the first matching device instance when selecting a file with a valid extension</string>
                </property>
                <property name="statusTip">
                 <string>Automatically select the first matching device instance when selecting a file with a valid extension</string>
                </property>
                <property name="text">
                 <string>Auto-select</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBoxChooserFilter">
                <property name="toolTip">
                 <string>Filter files by allowed extensions for the current device instance</string>
                </property>
                <property name="statusTip">
                 <string>Filter files by allowed extensions for the current device instance</string>
                </property>
                <property name="text">
                 <string>Filter</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLCDNumber" name="lcdNumberFileCounter">
                <property name="toolTip">
                 <string>Number of files scanned</string>
                </property>
                <property name="statusTip">
                 <string>Number of files scanned</string>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="numDigits">
                 <number>6</number>
                </property>
                <property name="digitCount">
                 <number>6</number>
                </property>
                <property name="segmentStyle">
                 <enum>QLCDNumber::Outline</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="toolButtonChooserPlay">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Play the selected configuration</string>
                </property>
                <property name="statusTip">
                 <string>Play the selected configuration</string>
                </property>
                <property name="icon">
                 <iconset resource="qmc2.qrc">
                  <normaloff>:/data/img/launch.png</normaloff>:/data/img/launch.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="toolButtonChooserPlayEmbedded">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Play the selected configuration (embedded)</string>
                </property>
                <property name="statusTip">
                 <string>Play the selected configuration (embedded)</string>
                </property>
                <property name="icon">
                 <iconset resource="qmc2.qrc">
                  <normaloff>:/data/img/embed.png</normaloff>:/data/img/embed.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QSplitter" name="splitterFileChooser">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <widget class="QTreeView" name="treeViewDirChooser">
              <property name="contextMenuPolicy">
               <enum>Qt::CustomContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Choose directory</string>
              </property>
              <property name="statusTip">
               <string>Choose directory</string>
              </property>
              <property name="rootIsDecorated">
               <bool>false</bool>
              </property>
              <property name="uniformRowHeights">
               <bool>true</bool>
              </property>
             </widget>
             <widget class="QTreeView" name="treeViewFileChooser">
              <property name="contextMenuPolicy">
               <enum>Qt::CustomContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Choose a file to map</string>
              </property>
              <property name="statusTip">
               <string>Choose a file to map</string>
              </property>
              <property name="rootIsDecorated">
               <bool>false</bool>
              </property>
              <property name="itemsExpandable">
               <bool>false</bool>
              </property>
              <property name="sortingEnabled">
               <bool>true</bool>
              </property>
              <property name="allColumnsShowFocus">
               <bool>true</bool>
              </property>
              <property name="expandsOnDoubleClick">
               <bool>false</bool>
              </property>
              <attribute name="headerShowSortIndicator" stdset="0">
               <bool>true</bool>
              </attribute>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBoxAvailableDeviceConfigurations">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Available device configurations</string>
      </property>
      <layout class="QGridLayout" name="gridLayoutAvailableDeviceConfigs">
       <property name="margin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QListWidget" name="listWidgetDeviceConfigurations">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="toolTip">
          <string>List of available device configurations</string>
         </property>
         <property name="statusTip">
          <string>List of available device configurations</string>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="6"/>
 <tabstops>
  <tabstop>lineEditConfigurationName</tabstop>
  <tabstop>toolButtonNewConfiguration</tabstop>
  <tabstop>toolButtonCloneConfiguration</tabstop>
  <tabstop>toolButtonRemoveConfiguration</tabstop>
  <tabstop>toolButtonSaveConfiguration</tabstop>
  <tabstop>tabWidgetDeviceSetup</tabstop>
  <tabstop>toolButtonConfiguration</tabstop>
  <tabstop>treeWidgetDeviceSetup</tabstop>
  <tabstop>treeWidgetSlotOptions</tabstop>
  <tabstop>comboBoxDeviceInstanceChooser</tabstop>
  <tabstop>checkBoxChooserAutoSelect</tabstop>
  <tabstop>checkBoxChooserFilter</tabstop>
  <tabstop>toolButtonChooserPlay</tabstop>
  <tabstop>toolButtonChooserPlayEmbedded</tabstop>
  <tabstop>treeViewDirChooser</tabstop>
  <tabstop>treeViewFileChooser</tabstop>
  <tabstop>listWidgetDeviceConfigurations</tabstop>
 </tabstops>
 <resources>
  <include location="qmc2.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>toolButtonConfiguration</sender>
   <signal>clicked()</signal>
   <receiver>toolButtonConfiguration</receiver>
   <slot>showMenu()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>28</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
